pkg/restful/resource.go

```diff
@@ -322,6 +322,8 @@ func (ar *AbstractResource) registerCustomRoutes() error {
 			routerFunc = ar.router.Put
 		case http.MethodDelete:
 			routerFunc = ar.router.Delete
+		default:
+			return errors.Errorf("Invalid method %s used in custom route", customRoute.Method)
```

Comment:
that would interpreted as a 500 error, what you are looking for here is `nuclio.NewErrMethodNotAllowed()`

Reply:
MethodNotAllowed (405) is a client error (4xx), when the client does something invalid - specifically calling a method he/she/it is not supposed to.
This is in the `registerCustomRoutes()` phase, upon initialization, pure server/code error (5xx).